cmd/southbound_main/southbound_main.go

```diff
@@ -15,47 +15,49 @@
 package main
 
 import (
+	"context"
 	"fmt"
-	"github.com/onosproject/onos-config/pkg/southbound"
-	"github.com/onosproject/onos-config/pkg/southbound/topocache"
 	"reflect"
 	"time"
 
+	"github.com/onosproject/onos-config/pkg/southbound"
+	"github.com/onosproject/onos-config/pkg/southbound/topocache"
+
 	"github.com/golang/protobuf/proto"
 	"github.com/openconfig/gnmi/client"
 )
 
 func main() {
 	device := topocache.Device{
-		Addr:     "localhost:10161",
-		Target:   "Test-onos-config",
+		Addr:   "localhost:10161",
+		Target: "Test-onos-config",
 		// Loaded from default-certificates.go
 		//CaPath:   "/Users/andrea/go/src/github.com/opennetworkinglab/onos-config/tools/test/devicesim/certs/onfca.crt",
 		//CertPath: "/Users/andrea/go/src/github.com/opennetworkinglab/onos-config/tools/test/devicesim/certs/client1.crt",
 		//KeyPath:  "/Users/andrea/go/src/github.com/opennetworkinglab/onos-config/tools/test/devicesim/certs/client1.key",
-		Timeout:  time.Second * 10,
+		Timeout: time.Second * 10,
 	}
-	target, err := southbound.GetTarget(southbound.Key{Key: device.Addr})
+	ctx := context.Background()
+	target := southbound.Target{}
+
+	_, err := target.ConnectTarget(ctx, device)
 	if err != nil {
-		fmt.Println("Creating device for addr: ", device.Addr)
-		target, _, err = southbound.ConnectTarget(device)
-		if err != nil {
```

Comment:
this if err != nil should be left there

Reply:
For which one? I have the err checking in the code for connectTarget but not for GetTarget. Do you mean handling the err for GetTarget?